### PR TITLE
feat: add fused stream implementation

### PR DESCRIPTION
**Pull request by nappa85**

### CODE REVIEW

**Review by lu-zero — APPROVED** [review]

File: src/stream.rs [hunk 1]
```diff
@@ -90,6 +91,16 @@ impl Stream for UEventsStream {
             }
         }
 
-        Poll::Pending
+        if matches!(this, Self::None) {
+            Poll::Ready(None)
+        } else {
+            Poll::Pending
```
Comment (dodomorandi) [inline src/stream.rs:97]: This should be unreachable, isn't it?

**Comment by nappa85** [comment]
No, now I set self to None also when the socket reads an empty buffer

Il mar 31 dic 2024, 15:26 Edoardo Morandi ***@***.***> ha
scritto:

**Comment by dodomorandi** [comment]
> No, now I set self to None also when the socket reads an empty buffer

Yes, but I mean, there is no way `this` does **not** match `None` at that point... or not?

**Comment by nappa85** [comment]
Ah, yes, the else part... Yes, this should be unreachable, I would prefer a
panic over debugging a future that is never waken

Il mar 31 dic 2024, 15:30 Edoardo Morandi ***@***.***> ha
scritto:

**Pull request merged**